codec-http2/src/main/java/io/netty5/handler/codec/http2/InboundHttpToHttp2Adapter.java

```diff
@@ -37,8 +38,11 @@ public InboundHttpToHttp2Adapter(Http2Connection connection, Http2FrameListener
     }
 
     private static int getStreamId(Http2Connection connection, HttpHeaders httpHeaders) {
-        return httpHeaders.getInt(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(),
-                                  connection.remote().incrementAndGetNextStreamId());
+        CharSequence streamId = httpHeaders.get(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text());
+        if (streamId != null) {
+            return Integer.parseInt(streamId.toString());
+        }
+        return connection.remote().incrementAndGetNextStreamId();
```

Comment:
same comment as in. https://github.com/netty/netty/pull/12735/files#diff-56499daa8bd2f57ca861aac556912e6f4d0d796694c9d4cb8e31c542cff7b6eeR80. I think this needs to be fixed in  4.1